libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayProgressObserver.kt

```diff
@@ -26,7 +26,7 @@ class ReplayProgressObserver @JvmOverloads constructor(
     private val replayRouteMapper: ReplayRouteMapper = ReplayRouteMapper()
 ) : RouteProgressObserver {
 
-    private var currentRouteLeg: RouteLeg? = null
```

Comment:
What was the leak? I can see how with this PR you've changed the logic a bit and optimized leg comparison, but what exactly was leaking?

Reply:
@dzinad , thanks for the question. I updated PR description. Does it make sense now?

Reply:
I see. So it's not technically a leak, we just use too much memory.